azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -262,12 +266,12 @@ public void run() {
     } catch (final Throwable t) {
       if (this.logger != null) {
         this.logger
-            .error(
-                "An error has occurred during the running of the flow. Quiting.",
-                t);
+            .error("An error has occurred during the running of the flow. Quiting.", t);
+      }
+      if (Status.KILLING.equals(this.flow.getStatus())) {
+        this.execMetrics.decrementFlowKillingCount();
       }
       this.flow.setStatus(Status.FAILED);
```

Comment:
I am curious whether this is just you trying to ensure FlowKillingCount stays accurate or have you in the past seen issues with Killing flows.

FlowRunner::kill() sets the status to `KILLING`. If we find the flow in Killing state, this indicates Killing itself failed. This is an interesting case. Since FlowRunner::kill() throws no exception, this can happen only in case of a Runtime Exception or java.lang.Error. Have you in the past seen 'Killing' resulting in these errors?

Reply:
I's essentially to keep the killing count dependable. I haven't seen that type of errors before.